### ISSUE TITLE
java.lang.ClassCastException when calling downloadSubtitles

### ISSUE THREAD

**Issue by calves07** [wtekiela/opensub4j#16]
Hello,

I have a SubtitleInfo object, but when I try to download it with 
```
List<SubtitleFile> subtitleFiles = osClient.downloadSubtitles(best.getId());
```
 I always get the following exception.

```
out 01, 2019 3:07:32 PM controller.DownloadSubtitlesController searchSubtitle
SEVERE: null
java.lang.ClassCastException: java.lang.String cannot be cast to [Ljava.lang.Object;
	at com.github.wtekiela.opensub4j.impl.ResponseParser.bindList(ResponseParser.java:26)
	at com.github.wtekiela.opensub4j.impl.ListOperation.execute(ListOperation.java:28)
	at com.github.wtekiela.opensub4j.impl.OpenSubtitlesClientImpl.downloadSubtitles(OpenSubtitlesClientImpl.java:183)
	at controller.DownloadSubtitlesController.searchSubtitle(DownloadSubtitlesController.java:50)
	at prometheus.Main.main(Main.java:37)
```

Any thoughts?

**Comment by wtekiela** [comment]
Probably duplicate of #11 

Error handling needs to be improved (contributions welcome) in case of wrong id.

I am not sure what `best` variable is in your snippet, but assuming it's an instance of `SubtitleInfo` class, then if you call:
```
osClient.downloadSubtitles(subtitleInfo.getSubtitleFileId());
```
instead of
```
osClient.downloadSubtitles(subtitleInfo.getId());
```
your code should work fine.

@calves07 please confirm.

**Comment by calves07** [comment]
Yeah it works now, thanks!

**Comment by wtekiela** [comment]
Reopening - I will enhance error reporting for this case in this issue.

**Comment by wtekiela** [comment]
Fixed ClassCastException and released in 0.2.3 - you would get empty list now, and response status in log files.

In 0.3 branch I will change the API to return ListResponse object containing the status as well.
